<launch>
    <param name='robot_description' textfile='$(find ls01b_v2)/urdf/cona.urdf'/>#
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
    
    
    <node name="ls01b_v2_1" pkg="ls01b_v2" type="ls01b_v2">
        
        <param name="scan_topic" value="scan1"/>         #设置激光数据topic名称
        <param name="frame_id" value="laser1"/>     #激光坐标
        <param name="serial_port" value="/dev/Lidar1"/>  #雷达连接的串口
        <param name="baud_rate" value="460800" />        #雷达连接的串口波特率
        <param name="angle_resolution" value="0.25"/>     #雷达角度分辨率
        <param name="zero_as_max" value="false"/>        # 设置为true探测不到区域会变成最大值
        <param name="min_as_zero" value="false"/>        # true：探测不到区域为0，false：探测不到区域为inf
        <param name="angle_disable_min_0" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_0" value="-1"/>
        <param name="angle_disable_min_1" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_1" value="-1"/>
        <param name="angle_disable_min_2" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_2" value="-1"/>
        <param name="angle_disable_min_3" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_3" value="-1"/>
        <param name="angle_disable_min_4" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_4" value="-1"/>
        <param name="robot_radius" value="0.2"/>         # 机器人半径
        <param name="center_x" value="0"/>            # 雷达坐标系下机器人中心位置
        <param name="center_y" value="0"/>
        <param name="min_range" value="0.2"/>
        <param name="max_range" value="25.0"/>
    </node>
    <node name="ls01b_v2_2" pkg="ls01b_v2" type="ls01b_v2">

        <param name="scan_topic" value="scan2"/>         #设置激光数据topic名称
        <param name="frame_id" value="laser2"/>     #激光坐标
        <param name="serial_port" value="/dev/Lidar2"/>  #雷达连接的串口
        <param name="baud_rate" value="460800" />        #雷达连接的串口波特率
        <param name="angle_resolution" value="0.25"/>     #雷达角度分辨率
        <param name="zero_as_max" value="false"/>        # 设置为true探测不到区域会变成最大值
        <param name="min_as_zero" value="false"/>        # true：探测不到区域为0，false：探测不到区域为inf
        <param name="angle_disable_min_0" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_0" value="-1"/>
        <param name="angle_disable_min_1" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_1" value="-1"/>
        <param name="angle_disable_min_2" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_2" value="-1"/>
        <param name="angle_disable_min_3" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_3" value="-1"/>
        <param name="angle_disable_min_4" value="-1"/>    # 角度制，从angle_disable_min到angle_disable_max之前的值为0
        <param name="angle_disable_max_4" value="-1"/>
        <param name="robot_radius" value="0.2"/>         # 机器人半径
        <param name="center_x" value="0"/>            # 雷达坐标系下机器人中心位置
        <param name="center_y" value="0"/>
        <param name="min_range" value="0.2"/>
        <param name="max_range" value="25.0"/>
    </node>


    <include file="/home/cona/test_ws/src/ros_astra_camera/launch/stereo_s_u3.launch"/>

    <arg name="rvizconfig" default="$(find urdf_tutorial)/rviz/urdf.rviz" />
    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/> -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" />
    

</launch>

